### PR TITLE
Replace snarkyjs mentions across articles to o1js

### PR DESCRIPTION
**Pull request by qwadratic**

### CODE REVIEW

File: docs/zkapps/o1js-reference/README.md [hunk 1]
```diff
@@ -32,6 +32,6 @@ See the [Contributing guidelines](https://github.com/o1-labs/o1js/blob/main/CONT
 
 High-quality community packages from open source developers are available for your project.
 
-- **snarkyjs-elgamal** A partially homomorphic encryption library for o1js based on Elgamal encryption: [GitHub](https://github.com/Trivo25/snarkyjs-elgamal) and [npm](https://www.npmjs.com/package/snarkyjs-elgamal)
```
Comment (barriebyron) [inline docs/zkapps/o1js-reference/README.md:35]: per https://github.com/o1-labs/docs2/blob/main/CONTRIBUTING.md#o1js-reference this page is auto-generated from code https://github.com/o1-labs/o1js/blob/main/README.md and updates with the next o1js release (can you remove this file from the PR please?)

File: docs/zkapps/tutorials/02-private-inputs-hash-functions.mdx [hunk 2]
```diff
@@ -39,7 +39,7 @@ In this tutorial, you build a smart contract with a piece of private state that
 
 ## Prerequisites
 
-This tutorial has been tested with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.9.0` and [SnarkyJS](https://www.npmjs.com/package/snarkyjs) `0.11.0`.
+This tutorial has been tested with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.13.0` and [o1js](https://www.npmjs.com/package/o1js) `0.12.1`.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/02-private-inputs-hash-functions.mdx:42]: if you have tested with 12.1, great! We have not officially tested with o1js, so the SnarkyJS release 11.0 is accurate

File: docs/zkapps/tutorials/03-deploying-to-a-network.mdx [hunk 3]
```diff
@@ -34,7 +34,7 @@ Mina zkApps are available only on feature-complete Berkeley, Mina's public Testn
 
 ## Prerequisites
 
-This tutorial has been tested with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.10.0` and [SnarkyJS](https://www.npmjs.com/package/snarkyjs) `0.11.3`.
+This tutorial has been tested with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.13.0` and [o1js](https://www.npmjs.com/package/o1js) `0.12.1`.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/03-deploying-to-a-network.mdx:37]: if you have tested with 12.1, great! We have not officially tested with o1js, so the SnarkyJS release 11.0 is accurate

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 4]
```diff
@@ -41,7 +41,7 @@ The [example project](https://github.com/o1-labs/docs2/tree/main/examples/zkapps
 
 ## Prerequisites
 
-This tutorial has been verified with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.11.0` and [SnarkyJS](https://www.npmjs.com/package/snarkyjs) `0.12.1`.
+This tutorial has been verified with [Mina zkApp CLI](https://github.com/o1-labs/zkapp-cli) version `0.13.0` and [o1js](https://www.npmjs.com/package/o1js) `0.12.1`.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:44]: if you have tested with 13.0, great! We have not officially tested with o1js, so the SnarkyJS release is accurate

File: docs/zkapps/experimental.mdx [hunk 5]
```diff
@@ -26,7 +26,7 @@ We appreciate any and all feedback you want to provide.
 
 The best place to provide feedback and ask questions is on [Mina Protocol Discord](https://bit.ly/MinaDiscord). 
 
-To ask zkApps questions and engage with other developers building zkApps with SnarkyJS, use the [#zkapps-developers](https://discord.com/channels/484437221055922177/915745847692636181) channel.
+To ask zkApps questions and engage with other developers building zkApps with o1js, use the [#zkapps-developers](https://discord.com/channels/484437221055922177/915745847692636181) channel.
```
Comment (barriebyron) [inline docs/zkapps/experimental.mdx:29]: I learned something new! I searched only on the files that show in the sidebar... you found an instance I had missed, thank you

File: docs/zkapps/tutorials/05-common-types-and-functions.mdx [hunk 6]
```diff
@@ -301,7 +301,7 @@ import {
   ...
   MerkleTree,
   ...
-} from 'snarkyjs'
+} from 'o1js'
```
Comment (barriebyron) [inline docs/zkapps/tutorials/05-common-types-and-functions.mdx:304]: good catch!

File: docs/zkapps/tutorials/11-advanced-account-updates.mdx [hunk 7]
```diff
@@ -13,7 +13,7 @@ zkApps can now be deployed to Berkeley Testnet.
 
 :::note
 
-This tutorial was last tested with [SnarkyJS](https://www.npmjs.com/package/snarkyjs) 0.9.6.
+This tutorial was last tested with [o1js](https://www.npmjs.com/package/o1js) 0.12.1.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/11-advanced-account-updates.mdx:16]: if you have tested with 12.1, great! We have not officially tested with o1js, so the SnarkyJS release is accurate

File: docs/zkapps/tutorials/interacting-with-zkapps-server-side.mdx [hunk 8]
```diff
@@ -12,7 +12,7 @@ zkApp programmability is not yet available on the Mina Mainnet. You can get star
 
 :::note
 
-This tutorial was last tested with [SnarkyJS](https://www.npmjs.com/package/snarkyjs) 0.8.0.
+This tutorial was last tested with [o1js](https://www.npmjs.com/package/o1js) 0.12.1.
```
Comment (barriebyron) [inline docs/zkapps/tutorials/interacting-with-zkapps-server-side.mdx:15]: if you have tested with 12.1, great! We have not officially tested with o1js, so the SnarkyJS release is accurate

**Review by barriebyron — CHANGES_REQUESTED** [review]
Thanks so much for your docs contributions @qwadratic 

See https://github.com/o1-labs/docs2/blob/main/CONTRIBUTING.md#o1js-reference The zkApp Developers > [o1js Reference](https://docs.minaprotocol.com/zkapps/o1js-reference) docs are generated content and are outside the scope of the Mina Docs repository. (thanks anyway, but please remove docs/zkapps/o1js-reference/README.md from the PR 

The o1js README https://github.com/o1-labs/o1js/blob/main/README.md (which populates the npm package page https://www.npmjs.com/package/o1js) is correct (yay) and will update the auto-generated published doc on the next release

If you have verified the tutorials with the updated versions of o1js, please indicate and we can update the versions, otherwise, we'll need to leave the verified SnarkyJS versions 

Questions? Please ask.

File: docs/zkapps/o1js-reference/README.md (hunk 1 above)
Comment (qwadratic) [reply docs/zkapps/o1js-reference/README.md]: done

**Comment by qwadratic** [comment]
@barriebyron I confirm that tutorials have been tested with latest zkapp-cli and o1js 0.12.1. Also the dependencies for these tutorials have o1js 0.12.1 as well.

**Comment by barriebyron** [comment]
@qwadratic  you are on fire! 
Can you verify that you have tested with o1js 13.0? If yes, we can update the version and text. Thank you for ALL of your actions

**Comment by qwadratic** [comment]
@barriebyron no I have not tested with 0.13. But 0.14 is released and I think we will update and test samples again soon.

File: docs/zkapps/experimental.mdx (hunk 5 above)
Comment (barriebyron) [reply docs/zkapps/experimental.mdx:29]: fixed in https://github.com/o1-labs/docs2/pull/861

**Comment by barriebyron** [comment]
thank you @qwadratic  for your updates, these fixes are published from other PRs, so closing this one 

Keep on keeping on, we appreciate your helping us show up at our best!

**Pull request closed without merging**